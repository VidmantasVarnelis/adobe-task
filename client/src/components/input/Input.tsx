import React from 'react';
import styled from 'styled-components';
import { IInput } from './input.types';
import Arrow from 'assets/arrow.svg';

const Container = styled.div`
  position: relative;
`;

const IconContainer = styled.button`
  position: absolute;
  right: 0;
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  width: 3rem;
  cursor: pointer;
  border: none;
  background-color: transparent;
`;
const Icon = styled.img`
  width: 1.5rem;
  height: 1.5rem;
`;
const InputComponent = styled.input`
  padding: 0.75rem 3rem 0.75rem 1rem;
  width: 100%;
  font-size: 1.2rem;
  border-radius: 0.25rem;
  border: none;
  &:focus {
    outline: 3px solid #588157;
  }
`;

const Input: React.FC<IInput> = ({ onChange, placeholder }) => {
  return (
    <Container>
      <IconContainer role="button">
        <Icon src={Arrow} alt="No icon" />
      </IconContainer>
      <InputComponent
        onChange={onChange}
        placeholder={placeholder}
        aria-label="number-input"
      />
    </Container>
  );
};

export default Input;
