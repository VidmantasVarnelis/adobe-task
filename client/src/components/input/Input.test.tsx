import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import Input from './Input';

describe('Input component Test Suite', () => {
  it('check if input correctly renders', () => {
    const { container } = render(<Input onChange={jest.fn()} />);
    expect(container).toMatchSnapshot();
  });
});
