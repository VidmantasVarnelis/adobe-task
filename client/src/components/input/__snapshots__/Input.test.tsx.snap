// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component Test Suite check if input correctly renders 1`] = `
<div>
  <div
    class="sc-bczRLJ ehiAyy"
  >
    <button
      class="sc-gsnTZi fskANS"
      role="button"
    >
      <img
        alt="No icon"
        class="sc-dkzDqf dRzneU"
        src=""
      />
    </button>
    <input
      aria-label="number-input"
      class="sc-hKMtZM Fowqy"
    />
  </div>
</div>
`;
